lib/engine/game/g_18_esp/game.rb

```diff
@@ -401,7 +405,7 @@ def setup_corporations
             hex.tile.remove_reservation!(c)
             hex.tile.cities[c.city || 0].remove_tokens!
             destination_hex = @hexes.find { |h| h.id == c.destination }
-            destination_hex.tile.icons = destination_hex.tile.icons.dup.reject { |icon| icon.name == c.name } if destination_hex
+            destination_hex.tile.icons = destination_hex.tile.icons.reject! { |icon| icon.name == c.name } if destination_hex
```

Comment:
`Array.reject!` modifies the array, you do not need to reassign it.
```suggestion
            destination_hex.tile.icons.reject! { |icon| icon.name == c.name } if destination_hex
```
This needs to be changed in the other places too.